public/app/features/transformers/suggestionsInput/SuggestionsInput.tsx

```diff
@@ -0,0 +1,207 @@
+      }
+    },
+    [showingSuggestions, suggestions, suggestionsIndex, onVariableSelect]
+  );
+
+  const onValueChanged = React.useCallback((event: FormEvent<HTMLInputElement>) => {
+    setVariableValue(event.currentTarget.value);
+  }, []);
+
+  const onBlur = React.useCallback(
+    (event: FormEvent<HTMLInputElement>) => {
+      onChange(event.currentTarget.value);
+    },
+    [onChange]
+  );
+
+  useEffect(() => {
+    setInputHeight(inputRef.current!.clientHeight);
+  }, []);
+
+  return (
+    <div className={styles.inputWrapper}>
+      {showingSuggestions && (
+        <Portal>
+          <ReactPopper
+            referenceElement={inputRef.current!}
+            placement="bottom-start"
+            modifiers={[
+              {
+                name: 'preventOverflow',
+                enabled: true,
+                options: {
+                  rootBoundary: 'viewport',
+                },
+              },
+              {
+                name: 'arrow',
+                enabled: false,
+              },
+              {
+                name: 'offset',
+                options: {
+                  offset: [0, 0],
+                },
+              },
+            ]}
+          >
+            {({ ref, style, placement }) => {
+              return (
+                <div ref={ref} style={style} data-placement={placement} className={styles.suggestionsWrapper}>
+                  <CustomScrollbar
+                    scrollTop={scrollTop}
+                    autoHeightMax="300px"
+                    setScrollTop={({ scrollTop }) => setScrollTop(scrollTop)}
+                  >
+                    {/* This suggestion component has a specialized name,
+                          but is rather generalistic in implementation,
+                          so we're using it in transformations also. 
+                          We should probably rename this to something more general. */}
+                    <DataLinkSuggestions
+                      activeRef={activeRef}
+                      suggestions={suggestions}
+                      onSuggestionSelect={onVariableSelect}
+                      onClose={() => setShowingSuggestions(false)}
+                      activeIndex={suggestionsIndex}
+                    />
+                  </CustomScrollbar>
+                </div>
+              );
+            }}
+          </ReactPopper>
+        </Portal>
+      )}
+      {invalid && error && (
+        <div className={styles.errorTooltip}>
+          <FieldValidationMessage>{error}</FieldValidationMessage>
+        </div>
+      )}
+      <Input
+        placeholder={placeholder}
+        invalid={invalid}
+        ref={inputRef}
+        value={variableValue}
+        onChange={onValueChanged}
+        onBlur={onBlur}
+        onKeyDown={onKeyDown}
+      />
+    </div>
+  );
+};
+
+SuggestionsInput.displayName = 'SuggestionsInput';
+
+function getElementPosition(suggestionElement: HTMLElement | null, activeIndex: number) {
```

Comment:
This one seems more like a library function. Maybe one for grafana-ui?

Reply:
I think it might be a bit too specific since it assumes all the elements are of equal size.